### ISSUE TITLE
Data types in VAE examples

### ISSUE THREAD

**Issue by jan-matthis** [blei-lab/edward#566]
Since 4395319a8442bbfbeb1da1f2c7c25d879a09be7f, the VAE examples are broken:

```
TypeError: Key-value pair in data does not have same dtype: <dtype: 'int32'>, <dtype: 'float32'>
```

.. because `x` is `Bernoulli` with dtype int32, whereas the dtype of `x_ph` is float32.

Changing placeholder and data to int32 will cause further errors downstream. Defining a new variable that casts `x` to `float32` and using that for the key works fine -- but I'm not sure if that's the best solution.

**Comment by dustinvtran** [comment]
Great catch. As you mention,  replacing
```python
x_ph = tf.placeholder(tf.float32, [M, 28 * 28])
```
with
```python
x_ph = tf.placeholder(tf.int32, [M, 28 * 28])
hidden = Dense(256, activation='relu')(tf.cast(x_ph, tf.float32))
```
fixes it. I think this is a good solution. The casting is necessary due to the way Keras's dense layer works, and is not so much an Edward problem. Happy to merge a PR for this, including any other examples where this occurs.

**Comment by dustinvtran** [comment]
On second thought, some work also needs to be done about the fact that `x_train` is a batch of pixel intensity values (between 0 and 1), whereas `x_ph` is now an integer placeholder which should only accept 0 and 1.

**Comment by dustinvtran** [comment]
Fixed in #571.
